src/routes/connect/+page.svelte

```diff
@@ -298,6 +315,37 @@
         </div>
       {/if}
 
+      <div class="w-full">
+        <p class="text-neutral-600 dark:text-neutral-300">
+          {$translate('app.inputs.add_rune.recipes')}
+        </p>
+        <div class="flex justify-between">
+          {#each recipes as recipe, i}
+            <div
+              on:click={handleCopy(createRuneRecipe(recipe, sessionPublicKey))}
+              class="relative flex items-center w-full my-4"
+            >
+              <span class="font-semibold">{$translate(`app.inputs.add_rune.${recipe}`)}</span>
+
+              <div
+                class:text-utility-success={copySuccess ===
+                  createRuneRecipe(recipe, sessionPublicKey)}
+              >
+                {#if copySuccess === createRuneRecipe(recipe, sessionPublicKey)}
```

Comment:
Even though the cost of recomputing the rune recipe string here is small, it all adds up and I think we can easily avoid recomputing for these comparisons.

First we can refactor the `handleCopy` function to take two arguments, a required `key` and then an optional `value`. It will use the `key` to set the `copySuccess` var so then all further comparisons will not require re-deriving the key:

```javascript
function handleCopy(key: string, value?: string) {
    return async () => {
      // write the `value` to the clipboard if available, otherwise the `key` as the value to write
      const success = await writeClipboardValue(value || key)

      if (success) {
        copySuccess = key

        copyAnimationTimeout = setTimeout(() => (copySuccess = ''), 3000)
      }
    }
  }
```

Then we can make those comparisons simply compare to the key that was set:

```javascript
<div class="flex justify-between">
          {#each recipes as recipe}
            <div
              on:click={handleCopy(recipe, createRuneRecipe(recipe, sessionPublicKey))}
              class="relative flex items-center w-full my-4"
            >
              <span class="font-semibold">{$translate(`app.inputs.add_rune.${recipe}`)}</span>

              <div class:text-utility-success={copySuccess === recipe}>
                {#if copySuccess === recipe}
                  <div in:fade class="w-8">
                    {@html check}
                  </div>
                {:else}
                  <div in:fade class="w-8">
                    {@html copy}
                  </div>
                {/if}
              </div>
            </div>
          {/each}
        </div>
```